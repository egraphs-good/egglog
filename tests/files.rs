use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
