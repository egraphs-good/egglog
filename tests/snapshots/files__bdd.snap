---
source: tests/files.rs
expression: serialized
---
nodes:
  b2-0:
    op: b2
    children: []
    eclass: "21"
    cost: 9223372036854776000
  b123-0:
    op: b123
    children: []
    eclass: "6"
    cost: 9223372036854776000
  b11-0:
    op: b11
    children: []
    eclass: "12"
    cost: 9223372036854776000
  or-3713696389113654829:
    op: or
    children:
      - b2-0
      - b11-0
    eclass: "23"
    cost: 1
  or-10384174419947986705:
    op: or
    children:
      - b1-0
      - or-3713696389113654829
    eclass: "24"
    cost: 1
  b0-0:
    op: b0
    children: []
    eclass: "2"
    cost: 9223372036854776000
  or121-0:
    op: or121
    children: []
    eclass: "24"
    cost: 9223372036854776000
  True-0:
    op: "True"
    children: []
    eclass: "15"
    cost: 9223372036854776000
  False-0:
    op: "False"
    children: []
    eclass: "19"
    cost: 9223372036854776000
  b12-0:
    op: b12
    children: []
    eclass: "18"
    cost: 9223372036854776000
  b1-0:
    op: b1
    children: []
    eclass: "12"
    cost: 9223372036854776000
root_eclasses: []
class_data:
  "21":
    type: BDD
  "6":
    type: BDD
  "12":
    type: BDD
  "23":
    type: BDD
  "24":
    type: BDD
  "2":
    type: BDD
  "15":
    type: BDD
  "19":
    type: BDD
  "18":
    type: BDD

